designer/server/src/main/scala/pl/touk/nussknacker/ui/statistics/UsageStatisticsReportsSettingsDeterminer.scala

```diff
@@ -46,20 +54,72 @@ object UsageStatisticsReportsSettingsDeterminer {
       processService
         .getLatestProcessesWithDetails(
           ScenarioQuery.unarchived,
-          GetScenarioWithDetailsOptions.detailsOnly.withFetchState
+          GetScenarioWithDetailsOptions.withsScenarioGraph.withFetchState
         )(user)
         .map { scenariosDetails =>
           scenariosDetails.map(scenario =>
             ScenarioStatisticsInputData(
               scenario.isFragment,
               scenario.processingMode,
               deploymentManagerTypeByProcessingType(scenario.processingType),
-              scenario.state.map(_.status)
+              scenario.state.map(_.status),
+              scenario.scenarioGraph.map(_.nodes.length).getOrElse(0),
+              scenario.processCategory,
+              scenario.processVersionId,
+              scenario.createdBy,
+              scenario.scenarioGraph
+                .map { graph =>
+                  graph.nodes.map {
+                    case _: FragmentInput => 1
+                    case _                => 0
+                  }.sum
+                }
+                .getOrElse(0)
             )
           )
         }
     }
-    new UsageStatisticsReportsSettingsDeterminer(config, fingerprintSupplier, fetchNonArchivedScenarioParameters)
+    def fetchActivity(): Future[List[DbProcessActivityRepository.ProcessActivity]] = {
+      implicit val user: LoggedUser = NussknackerInternalUser.instance
+      for {
+        scenarioDetails <- processService.getLatestProcessesWithDetails(
+          ScenarioQuery.unarchived,
+          GetScenarioWithDetailsOptions.detailsOnly.withFetchState
+        )
+        activity <- scenarioDetails.map { scenario =>
+          scenarioActivityRepository.findActivity(scenario.processIdUnsafe)
+        }.sequence
+      } yield activity
+    }
+
+    def fetchComponentList(): Future[List[ComponentListElement]] = {
+      implicit val user: LoggedUser = NussknackerInternalUser.instance
+      componentService.getComponentsList
+    }
+
+    def fetchScenarioActions(): Future[List[List[ProcessAction]]] = {
+      implicit val user: LoggedUser = NussknackerInternalUser.instance
+      processService
+        .getLatestProcessesWithDetails(
```

Comment:
We invoke `processService.getLatestProcessesWithDetails` for the 3 times, maybe we could invoke it only once and then filter for apropriate elements?

Reply:
I see the `getLatestProcessesWithDetails` in `fetchActivity` and `fetchNonArchivedScenarioParameters`, maybe we can `processId` to `ScenarioStatisticsInputData` and then pass to `fetchActivity(scenariosInputData)`. What do you think?